src/lib.rs

```diff
@@ -440,6 +446,10 @@ pub trait Handler {
     }
 
     fn halt_reason(&self) -> Result<StopReason, Error>;
+
+    fn invoke(&self, &[u8]) -> Result<String, Error> {
```

Comment:
Going forward it might be good to stick at least a cursory doc comment on `Handler` methods so that we don't have to go back after the fact and document everything from scratch, especially considering that this is going to be the largest part of the API for users of this crate.

Reply:
I did some of this in a subsequent patch, but there is still more to do.